homeassistant/components/smartthings/__init__.py

```diff
@@ -357,31 +359,28 @@ def any_assigned(self, device_id: str, platform: str):
                 data=evt.data,
             )
 
-            # Fire events for buttons
+            data = {
+                "location_id": evt.location_id,
+                "device_id": evt.device_id,
+                "component_id": evt.component_id,
+                "name": device.label,
+                "capability": evt.capability,
+                "attribute": evt.attribute,
+                "value": evt.value,
+                "data": evt.data,
+            }
             if (
                 evt.capability == Capability.button
                 and evt.attribute == Attribute.button
             ):
-                data = {
-                    "component_id": evt.component_id,
-                    "device_id": evt.device_id,
-                    "location_id": evt.location_id,
-                    "value": evt.value,
-                    "name": device.label,
-                    "data": evt.data,
-                }
+                # Fire events for buttons
                 self._hass.bus.async_fire(EVENT_BUTTON, data)
                 _LOGGER.debug("Fired button event: %s", data)
+            elif self._entry.options.get(CONF_RAISE_EVENTS):
```

Comment:
Why do we need an option? Can't we just fire the event? The user will not be required to create an automation to listen for it. It's already optional.

Reply:
I figured we didn’t want to push everything to the event bus if we didn’t need to.  If you don’t think it matters, then I can simplify it.

Reply:
I think it's ok to fire this event.